02_bubble/amitkrishna/bubble_test.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var sortingTestsData = []struct {
+	n        []int // input
+	expected []int // expected result
+}{
+	{[]int{3, 2, 1, 5}, []int{1, 2, 3, 5}},
+	{[]int{3, 2, 1, 5, 8}, []int{1, 2, 3, 5, 8}},
+}
+
+func TestBubbleSortMainOutput(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := "[1 2 3 5]"
+	actual := buf.String()
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+
+func TestBubbleSortOutput(t *testing.T) {
+	r := require.New(t)
+	for _, tt := range sortingTestsData {
+		actual := bubble(tt.n)
+		r.ElementsMatchf(tt.expected, actual, "Unexpected output in main()")
```

Comment:
Sorry I had this wrong, just [require.Equalf](https://godoc.org/github.com/stretchr/testify/require#Equalf) is the best option.